sympy/printing/maple.py

```diff
@@ -0,0 +1,272 @@
+
+    def _print_NumberSymbol(self, expr):
+        return number_symbols[expr]
+
+    def _print_NegativeInfinity(self, expr):
+        return '-infinity'
+
+    def _print_Infinity(self, expr):
+        return 'infinity'
+
+    def _print_Idx(self, expr):
+        return self._print(expr.label)
+
+    def _print_BooleanTrue(self, expr):
+        return "true"
+
+    def _print_BooleanFalse(self, expr):
+        return "false"
+
+    def _print_bool(self, expr):
+        return str(expr).lower()
+
+    def _print_NaN(self, expr):
+        return 'undefined'
+
+    def _get_matrix(self, expr, sparse=False):
+        if expr.cols == 0 or expr.rows == 0:
+            _strM = 'Matrix([], storage = {storage})'.format(
+                storage='sparse' if sparse else 'rectangular')
+        else:
+            _strM = 'Matrix({list}, storage = {storage})'.format(
+                list=self._print(expr.tolist()),
+                storage='sparse' if sparse else 'rectangular')
+        return _strM
+
+    def _print_MatrixElement(self, expr):
+        return "{parent}[{i_maple}, {j_maple}]".format(
+            parent=self.parenthesize(expr.parent, PRECEDENCE["Atom"], strict=True),
+            i_maple=self._print(expr.i + 1),
+            j_maple=self._print(expr.j + 1))
+
+    def _print_MatrixBase(self, expr):
+        return self._get_matrix(expr, sparse=False)
+
+    def _print_SparseMatrix(self, expr):
+        return self._get_matrix(expr, sparse=True)
+
+    _print_Matrix = \
+        _print_DenseMatrix = \
+        _print_MutableDenseMatrix = \
+        _print_ImmutableMatrix = \
+        _print_ImmutableDenseMatrix = \
+        _print_MatrixBase
+    _print_MutableSparseMatrix = \
+        _print_ImmutableSparseMatrix = \
+        _print_SparseMatrix
+
+    def _print_Identity(self, expr):
+        return self._print(sympy.SparseMatrix(expr))
+
+    def _print_MatMul(self, expr):
+        _fact_list = list(expr.args)
+        _const = None
+        if not (
+            isinstance(_fact_list[0], sympy.MatrixBase) or isinstance(
+            _fact_list[0], sympy.MatrixExpr) or isinstance(
+            _fact_list[0], sympy.MatrixSlice) or isinstance(
+            _fact_list[0], sympy.MatrixSymbol)):
+            _const, _fact_list = _fact_list[0], _fact_list[1:]
+
+        if _const is None:
+            return '.'.join(self._print(_m) for _m in _fact_list)
+        else:
+            return '{c}*{m}'.format(c=_const, m='.'.join(self._print(_m) for _m in _fact_list))
+
+    def _print_MatPow(self, expr):
+        # This function requires LinearAlgebra Function in Maple
+        return 'MatrixPower({A}, {n})'.format(A=expr.base, n=expr.exp)
+
+    def _print_HadamardProduct(self, expr):
+        _fact_list = list(expr.args)
+        return '*'.join(self._print(_m) for _m in _fact_list)
+
+
+
+def maple_code(expr, assign_to=None, **settings):
+    r"""Converts `expr` to a string of Maple code.
+
+        Parameters
```

Comment:
Update with the parameters that you (plan to) support. I think at least `inline` and `human` could go. `precision` should be OK (`Digits`). `contract` I have no idea. `assign_to` will be useful (maybe it is implemented, looks like it from the tests). `user_functions` you have already implemented, which is nice.

Reply:
@oscargus , Ok understood.